frame/balances/src/lib.rs

```diff
@@ -859,7 +852,11 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 				system::Pallet::<T>::dec_consumers(who);
 			}
 		} else {
-			Locks::<T, I>::insert(who, locks);
+			let bounded_locks = <BoundedVec<BalanceLock<T::Balance>, T::MaxLocks>>::force_from(
```

Comment:
Haha we need to fix this at some point, but yeah this looks good.

Reply:
What alternative do you have in mind? 

I tried to replicate the old code, and the old code was not strict, so I added this `force_from`.

Reply:
yeah the way this API is written historically is not strict. ideally something like `set_lock` would return a result, and we would check that wherever `set_lock` is used

But that is beyond the scope here.